src/org/labkey/test/tests/PostgresQueriesTest.java

```diff
@@ -0,0 +1,174 @@
+        verifyLocksGrid();
+        verifyApiAccess(true);
+
+        // log in as troubleshooter
+        signOut();
+        signIn(TROUBLESHOOTER_USER);
+        goToAdminConsole().clickPostgresActivity();
+        verifyActivityGrid(false);
+
+        goToAdminConsole().clickPostgresLocks();
+        verifyLocksGrid();
+        verifyApiAccess(true);
+    }
+
+    private void verifyApiAccess(boolean expectSuccess) throws IOException, CommandException
+    {
+        SelectRowsResponse activityResponse = selectRows(expectSuccess, "pg_stat_activity");
+        if (expectSuccess)
+        {
+            assertFalse("Should have at least one row", activityResponse.getRows().isEmpty());
+            assertNotNull("Should have a pid", activityResponse.getRows().get(0).get("pid"));
+        }
+        SelectRowsResponse locksResponse = selectRows(expectSuccess, "pg_locks");
+        if (expectSuccess)
+        {
+            assertFalse("Should have at least one row", locksResponse.getRows().isEmpty());
+            assertNotNull("Should have a pid", locksResponse.getRows().get(0).get("locktype"));
+        }
+    }
+
+    private SelectRowsResponse selectRows(boolean expectSuccess, String queryName) throws IOException, CommandException
+    {
+        Connection connection = WebTestHelper.getRemoteApiConnection();
+        SelectRowsCommand command = new SelectRowsCommand("postgres", queryName);
+        try
+        {
+            SelectRowsResponse response = command.execute(connection, "/");
+            if (!expectSuccess)
+            {
+                fail("Request should have failed");
+            }
+            return response;
+        }
+        catch (CommandException e)
+        {
+            if (expectSuccess)
+            {
+                throw e;
+            }
+            else
+            {
+                assertEquals("Wrong status code", 404, e.getStatusCode());
+            }
+        }
+        return null;
+    }
+
+    private void verifyLocksGrid()
+    {
+        assertTextPresent("pg_locks");
+        DataRegionTable table = new DataRegionTable("query", this);
+        List<String> cols = table.getColumnLabels();
+        assertTrue("Didn't find a Locktype column: " + cols, cols.contains("Locktype"));
+        assertTrue("Didn't find a Virtualtransaction column: " + cols, cols.contains("Virtualtransaction"));
+    }
+
+    private void verifyActivityGrid(boolean expectDelete)
+    {
+        assertTextPresent("pg_stat_activity");
+        DataRegionTable table = new DataRegionTable("query", this);
+        List<String> cols = table.getColumnLabels();
+        assertTrue("Didn't find a Query column: " + cols, cols.contains("Query"));
+        assertTrue("Didn't find a Running Time Ms column: " + cols, cols.contains("Running Time Ms"));
+        assertTextPresent("GREATEST(EXTRACT(MILLISECONDS FROM AGE(NOW(), query_start)), 0) END AS running_time_ms");
+        assertTextPresent("AsyncQueryRequest:");
```

Comment:
`assertTextPresent` takes multiple strings to avoid fetching the full page source multiple times.
```suggestion
        assertTextPresent(
            "GREATEST(EXTRACT(MILLISECONDS FROM AGE(NOW(), query_start)), 0) END AS running_time_ms", 
            "AsyncQueryRequest:");
```